pkg/controllermanager/controller/seed/seed_lifecycle_control.go

```diff
@@ -98,34 +105,54 @@ func (c *defaultControl) Reconcile(seedObj *gardencorev1beta1.Seed, key string)
 		seedLogger = logger.NewFieldLogger(logger.Logger, "seed", seed.Name)
 	)
 
-	// New seeds don't have conditions, i.e., gardenlet never reported anything yet. Let's wait until it sends the first heart beat.
+	// New seeds don't have conditions - gardenlet never reported anything yet. Wait for grace period.
 	if len(seed.Status.Conditions) == 0 {
 		return true, nil
 	}
 
-	for _, condition := range seed.Status.Conditions {
-		// If the `GardenletReady` condition is not yet `Unknown` then check when it most recently sent a heartbeat and wait for the
-		// configured `monitorPeriod` before proceeding with any action.
-		if condition.Type == gardencorev1beta1.SeedGardenletReady && condition.Status != gardencorev1beta1.ConditionUnknown && !condition.LastUpdateTime.UTC().Before(time.Now().UTC().Add(-c.config.Controllers.Seed.MonitorPeriod.Duration)) {
+	observedSeedLease, err := c.leaseLister.Leases(v1beta1constants.GardenSeedLeaseNamespace).Get(seed.Name)
+	if client.IgnoreNotFound(err) != nil {
+		return false, err
+	}
+
+	if observedSeedLease != nil && observedSeedLease.Spec.RenewTime != nil {
+		if observedSeedLease.Spec.RenewTime.After(time.Now().UTC().Add(-c.config.Controllers.Seed.MonitorPeriod.Duration)) {
+			return true, nil
+		}
+
+		latestLeaseObject := &coordinationv1.Lease{}
+		err = c.k8sGardenClient.Client().Get(ctx, kutil.Key(v1beta1constants.GardenSeedLeaseNamespace, seed.Name), latestLeaseObject)
```

Comment:
So this is to not accidentally make the wrong decision due to outdated cache, hence, we fetch it again from the system to double-check?

Reply:
Yes